### ISSUE TITLE
spike - investigate possible bug with unintended limiters

### ISSUE THREAD

**Issue by tclayton33** [emory-libraries/blacklight-catalog#1407]
Multiple users at the Theology library are reporting issues with Advanced search applying limiters that they did not choose. It seems to be an intermittent but recurring issue that is happening across different browsers, including incognito sessions. 

Several (but not all) examples involved the "language=latin" and a "title starts with" limiter being applied. Example - user searches for title=Marriage AND Authur=Reynolds, but search also applies Language=Latin AND Title starts w/ L.

<img width="1271" src="https://api.zenhub.com/attachedFiles/eyJfcmFpbHMiOnsibWVzc2FnZSI6IkJBaHBBOGhNQkE9PSIsImV4cCI6bnVsbCwicHVyIjoiYmxvYl9pZCJ9fQ==--2cfa483fd3115d5e338a4fef4f619aa1eb085be5/Advanced%20search%20and%20uninteded%20limiters1.png" alt="Advanced search and uninteded limiters1.png" />

Or the user is performing and advanced search for a local call number, such as "1511 GEIL" but the search adds a limiter for subject display ssim = Bible-periodicals-indexes.

<img width="2000" src="https://api.zenhub.com/attachedFiles/eyJfcmFpbHMiOnsibWVzc2FnZSI6IkJBaHBBOFZNQkE9PSIsImV4cCI6bnVsbCwicHVyIjoiYmxvYl9pZCJ9fQ==--03f91f27d54c29bc1dd8a8feee3e63fb4ec9d44c/advanced%20search%20unintended%20limiters2.png" alt="advanced search unintended limiters2.png" />

Users have been advised to clear previous searches by using the "start over" button. Bo states that "The “start over” option does not fix the problem. Manually removing each superfluous limiter by clicking the “x” next to it is the only way to get the results one wants

Other observations from Bo:

- The limiters that it adds are not always the same. I’ve attached screenshots here of a different set of limiters that I have encountered (and I’m certain I have never used these limiters deliberately in a search).

<img width="1649" src="https://api.zenhub.com/attachedFiles/eyJfcmFpbHMiOnsibWVzc2FnZSI6IkJBaHBBOGRNQkE9PSIsImV4cCI6bnVsbCwicHVyIjoiYmxvYl9pZCJ9fQ==--5d19d86df529af8da12fe9dcbe97745d3689bbf7/unintended%20limiters.png" alt="unintended limiters.png" />

- Several of our colleagues have experienced this while using other library systems (e.g. Aeon or ArchivesSpace), and I have experienced it both while logged in and when not logged in. I typically get to LibrarySearch by going to the Pitts homepage and then clicking LibrarySearch from our QuickLinks menu, but others have LibrarySearch bookmarked.
- the problem happens sporadically, but we have definitely observed it coming and going over the last 2 weeks
- Btw, I tested this morning, and the advanced search problem is happening again. See the attached screenshot, where a local call number search for “1511 GEIL” adds the limiter for subject display ssim. This is also happening in Chrome incognito. I am reaching Library Search through the Quick Link on the Pitts website and then clicking “Advanced Search.” 

So far, I have not been able to replicate this issue. We need to determine if this is a bug, user error, or something that needs an enhancement request. 

I've [created a document](https://emory.sharepoint.com/:w:/r/sites/EmoryUniversityLibraries/Shared%20Documents/Staff/Committees%20%26%20Working%20Groups/Library%20Search%20Committee/Working%20Documents/Issue%20Unintended%20filters%20appearing%20in%20advanced%20search.docx?d=w40c15ec6b87c45feb84634882fe26841&csf=1&web=1&e=KtGpE0) in the Library Search Committee's Sharepoint that we can use to collect more examples and information directly from library staff who are experiencing this issue.

**Comment by abelemlih** [comment]
This issue is likely due to irrelevant search data persisting between searches due to the caching we implemented in this [ticket](https://app.zenhub.com/workspaces/blacklight-catalog-5f5f84a8a6d29939a0bc5d78/issues/gh/emory-libraries/blacklight-catalog/1389). I deployed branch `abel-1407-fix_advanced_search` to Test to verify that disabling caching resolves all the issues currently seen in production. Disabling caching will lead to the same performance issues we have seen previously, so we will need to develop a new solution for that issue.
